writer_test.go

```diff
@@ -92,3 +92,29 @@ Bye.
 		t.Error("Invalid mbox output:", s)
 	}
 }
+
+// Test for the fix for https://github.com/emersion/go-mbox/issues/21
+func TestMultipleWrites(t *testing.T) {
+	var buffer bytes.Buffer
+	mboxWriter := NewWriter(&buffer)
+	w, err := mboxWriter.CreateMessage("-", time.Time{})
+	if err != nil {
+		panic(err)
+	}
+	b := []byte("some text   ")
+	n, err := w.Write(b)
+	if n != 12 {
+		t.Errorf("unexpected return value for write: %d (expected %d)", n, 12)
+	}
+	if len(b) != 12 {
```

Comment:
I don't think this can ever be hit? `len(b)` is not something that `Write` can change.

Maybe the check could be simplified to `n != len(b)`?

Reply:
Yeah, you're right, checking back the buffer length is stupid. Will change to what you suggest in v2.